front/form/form_renderer.php

```diff
@@ -58,8 +63,23 @@
     Response::sendError(404, __("Form not found"));
 }
 
-// TODO: if displaying a form, check form access configuration (not yet implemented)
-// TODO: if previewing a form, check view rights on forms
+$manager = FormAccessControlManager::getInstance();
+
+// Manual session checks depending on the configurated access
+// This is a separate step from the `canAnswerForm` step to force each
+// access controls to be explicit on this point and avoid opening
+// too much access to unauthenticated users by mistake.
+$firewall = new Firewall($CFG_GLPI['root_doc']);
+if ($manager->canUnauthenticatedUsersAccessForm($form)) {
+    $firewall->applyStrategy($_SERVER['PHP_SELF'], Firewall::STRATEGY_NO_CHECK);
+} else {
+    $firewall->applyStrategy($_SERVER['PHP_SELF'], Firewall::STRATEGY_AUTHENTICATED);
+}
```

Comment:
I see what you mean but wouldn't it mean that each implementation of `canAnswerForm` would be responsible for calling the Firewall ? 
I thought it would not be OK.

Reply:
You can call the methods `Session::check*()` directly, the firewall does not do so much.

Reply:
Yes but it exposes the internal session validation logic, which IMO plugins shouldn't have to manipulate.
Couldn't this lead to issues in the future if we want to change the firewall behavior ?